ddl/ddl_api.go

```diff
@@ -452,6 +458,76 @@ func checkDuplicateColumn(colDefs []*ast.ColumnDef) error {
 	return nil
 }
 
+func checkGeneratedColumn(colDefs []*ast.ColumnDef) error {
+	normalColNames := map[string]bool{}
+	dependColNames := make([]string, 0)
+	for _, colDef := range colDefs {
+		depCols, err := findDependedColumnNames(*colDef)
+		if err != nil {
+			return err
+		}
+		normalColNames[colDef.Name.Name.L] = true
+		dependColNames = append(dependColNames, depCols...)
+	}
+	return columnNamesCover(normalColNames, dependColNames)
+}
+
+// columnNamesCover checks dependColNames is coverd by normalColNames or not.
+// if not, return a formatted error.
+func columnNamesCover(normalColNames map[string]bool, dependColNames []string) error {
+	for _, name := range dependColNames {
+		if _, ok := normalColNames[name]; !ok {
+			return errBadField.GenByArgs(name, "generated column function")
+		}
+	}
+	return nil
+}
+
+// findDependedColumnNames returns a slice of string, which is depend by colDef.
```

Comment:
which indicates the names of the columns that are depended by colDef

will this be better